### PR TITLE
chapter 14 created

### PR DESCRIPTION
**Pull request by agustinaGallo97**
## :link: Jira Tracker reference:
*[AP - Chapter 14] The App Bar - https://androidtraining.atlassian.net/browse/AT-54

## Description:
Chapter 14 created

## :camera_flash: UI preview
ezgif com-video-to-gif(13)

### CODE REVIEW

File: app/src/main/java/com/bignerdranch/android/criminalintent/controllers/CrimeListFragment.kt [hunk 1]
```diff
@@ -55,10 +57,31 @@ class CrimeListFragment : Fragment(R.layout.fragment_crime_list) {
...
 
   private fun setUpCrimeRecyclerView(view: View) {
-    crimeRecyclerView = view.findViewById<RecyclerView>(R.id.crimeRecyclerView)
+    crimeRecyclerView = view.findViewById(R.id.crimeRecyclerView)
     crimeRecyclerView.layoutManager = LinearLayoutManager(context)
     crimeRecyclerView.adapter = adapter
   }
 
   private fun updateUI(crimes: List<Crime>) = adapter.submitList(crimes)
+  override fun onCreate(savedInstanceState: Bundle?) {
```
Comment (FerrariBruno) [inline app/src/main/java/com/bignerdranch/android/criminalintent/controllers/CrimeListFragment.kt:82]:
```suggestion

  override fun onCreate(savedInstanceState: Bundle?) {
```

File: app/src/main/java/com/bignerdranch/android/criminalintent/models/CrimeListViewModel.kt [hunk 2]
```diff
@@ -6,4 +6,7 @@ import com.bignerdranch.android.criminalintent.CrimeRepository
 class CrimeListViewModel : ViewModel() {
   private val crimeRepository = CrimeRepository.get()
   val crimeListLiveData = crimeRepository.getCrimes()
+
+  fun addCrime(crime: Crime) =
+    crimeRepository.addCrime(crime)
```
Comment (FerrariBruno) [inline app/src/main/java/com/bignerdranch/android/criminalintent/models/CrimeListViewModel.kt]: sl is fine

File: app/src/main/res/menu/fragment_crime_list.xml [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+    xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:app="http://schemas.android.com/apk/res-auto"
+    >
+    <item
+        android:id="@+id/newCrime"
+        android:icon="@drawable/ic_menu_add"
+        android:title="@string/new_crime"
+        app:showAsAction="ifRoom|withText"
+        />
+</menu>
```
Comment (FerrariBruno) [inline app/src/main/res/menu/fragment_crime_list.xml]:
```suggestion
</menu>

```

File: app/src/main/res/values/strings.xml [hunk 4]
```diff
@@ -8,4 +8,5 @@
     <string name="crime_date">Crime Date</string>
     <string name="crime_time">Crime Time</string>
     <string name="crime_title_pressed"> %1$s pressed!</string>
+    <string name="new_crime"> New crime</string>
```
Comment (FerrariBruno) [inline app/src/main/res/values/strings.xml:11]:
```suggestion
    <string name="new_crime">New crime</string>
```

**Review by FerrariBruno — APPROVED** [review]

File: app/src/main/java/com/bignerdranch/android/criminalintent/controllers/CrimeListFragment.kt [hunk 5]
```diff
@@ -49,16 +52,52 @@ class CrimeListFragment : Fragment(R.layout.fragment_crime_list) {
     ) { crimes ->
       crimes.let {
         Timber.d("Got crimes ${crimes.size}")
-        updateUI(crimes)
+        if (crimes.size == 0) {
```
Comment (FerrariBruno) [inline app/src/main/java/com/bignerdranch/android/criminalintent/controllers/CrimeListFragment.kt]: method?

File: app/src/main/res/values/strings.xml [hunk 6]
```diff
@@ -8,4 +8,7 @@
...
+    <string name="empty_list_crimes"> Empty list crimes</string>
```
Comment (FerrariBruno) [inline app/src/main/res/values/strings.xml]:
```suggestion
    <string name="empty_list_crimes">Empty list crimes</string>
```

File: app/src/main/res/values/strings.xml [hunk 7]
```diff
@@ -8,4 +8,7 @@
...
+    <string name="add_crime"> Add new crime</string>
```
Comment (FerrariBruno) [inline app/src/main/res/values/strings.xml]:
```suggestion
    <string name="add_crime">Add new crime</string>
```

**Review by FerrariBruno — APPROVED** [review]

File: app/src/main/java/com/bignerdranch/android/criminalintent/controllers/CrimeListFragment.kt [hunk 8]
```diff
@@ -49,16 +52,56 @@ class CrimeListFragment : Fragment(R.layout.fragment_crime_list) {
...
         Timber.d("Got crimes ${crimes.size}")
-        updateUI(crimes)
+        if (crimes.size == 0) setEmptyListView() else setListView(crimes)
       }
     }
   }
 
+  private fun setListView(crimes: List<Crime>) {
+    emptyCrimeListAdvert.setVisibility(View.GONE)
+    addCrimeButton.setVisibility(View.GONE)
```
Comment (mirland) [inline app/src/main/java/com/bignerdranch/android/criminalintent/controllers/CrimeListFragment.kt:62]: Use core extensions and kotlin way

File: app/src/main/java/com/bignerdranch/android/criminalintent/controllers/CrimeListFragment.kt [hunk 9]
```diff
@@ -49,16 +52,56 @@ class CrimeListFragment : Fragment(R.layout.fragment_crime_list) {
     ) { crimes ->
       crimes.let {
         Timber.d("Got crimes ${crimes.size}")
-        updateUI(crimes)
+        if (crimes.size == 0) setEmptyListView() else setListView(crimes)
```
Comment (mirland) [inline app/src/main/java/com/bignerdranch/android/criminalintent/controllers/CrimeListFragment.kt]: Use isempty

File: app/src/main/res/layout/fragment_crime_list.xml [hunk 10]
```diff
@@ -1,7 +1,33 @@
...
+
+    <Button
+        android:id="@+id/addCrimeButton"
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:text="@string/add_crime"
+        />
+
+    <androidx.recyclerview.widget.RecyclerView
+        xmlns:android="http://schemas.android.com/apk/res/android"
```
Comment (mirland) [inline app/src/main/res/layout/fragment_crime_list.xml:27]: Rm

File: app/src/main/res/menu/fragment_crime_list.xml [hunk 11]
```diff
@@ -0,0 +1,12 @@
+<?xml version="1.0" encoding="utf-8"?>
+<menu
+    xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:app="http://schemas.android.com/apk/res-auto"
+    >
+    <item
```
Comment (mirland) [inline app/src/main/res/menu/fragment_crime_list.xml:7]: Nl

File: app/src/main/res/menu/fragment_crime_list.xml (hunk 3 above)
Comment (mirland) [inline app/src/main/res/menu/fragment_crime_list.xml:14]: Nl

**Review by mirland — APPROVED** [review]

**Review by f7deleon — APPROVED** [review]

File: app/src/main/java/com/bignerdranch/android/criminalintent/controllers/CrimeListFragment.kt (hunk 8 above)
Comment (mirland) [reply app/src/main/java/com/bignerdranch/android/criminalintent/controllers/CrimeListFragment.kt:62]: use it !

File: app/src/main/java/com/bignerdranch/android/criminalintent/controllers/CrimeListFragment.kt [hunk 12]
```diff
@@ -49,16 +52,56 @@ class CrimeListFragment : Fragment(R.layout.fragment_crime_list) {
...
+    super.onCreate(savedInstanceState)
+    setHasOptionsMenu(true)
+  }
+
+  override fun onCreateOptionsMenu(menu: Menu, inflater: MenuInflater) {
+    super.onCreateOptionsMenu(menu, inflater)
+    inflater.inflate(R.menu.fragment_crime_list, menu)
+  }
+
+  override fun onOptionsItemSelected(item: MenuItem): Boolean {
```
Comment (mirland) [inline app/src/main/java/com/bignerdranch/android/criminalintent/controllers/CrimeListFragment.kt:92]: short return

**Pull request merged**